react/ModalState.js

```diff
@@ -0,0 +1,364 @@
+        return
+
+      case isErrorNopickupsState:
+        this.setActiveState(ERROR_NOT_FOUND)
+        return
+
+      default:
+        return
+    }
+  }
+
+  getInitialActiveState = props => {
+    return props.askForGeolocation
+      ? GEOLOCATION_SEARCHING
+      : props.selectedPickupPoint
+        ? DETAILS
+        : props.pickupPoints.length > 0
+          ? LIST
+          : ERROR_NOT_FOUND
+  }
+
+  isDifferentGeoCoords(a, b) {
+    return a[0] !== b[0] || a[1] !== b[1]
+  }
+
+  isCurrentState(state, activeState) {
+    return state === activeState
+  }
+
+  setMapCenterLatLng = lastMapCenterLatLng =>
+    this.setState({ lastMapCenterLatLng })
+
+  setAskForGeolocation = askForGeolocation =>
+    this.setState({ askForGeolocation })
+
+  setShouldSearchArea = shouldSearchArea => this.setState({ shouldSearchArea })
+
+  setGeolocationStatus = status => this.setState({ geolocationStatus: status })
+
+  setActiveState = state =>
+    this.setState({ lastState: this.state.activeState, activeState: state })
+
+  setActiveSidebarState = state => {
+    this.setState({
+      lastSidebarState: this.state.activeSidebarState,
+      activeSidebarState: state,
+    })
+  }
+
+  findSla = (li, pickupPoint) => {
+    return li.slas.find(simulationPickupPoint =>
+      simulationPickupPoint.id.includes(
+        pickupPoint && (pickupPoint.id || pickupPoint.pickupPointId)
+      )
+    )
+  }
+
+  setSelectedPickupPoint = pickupPoint => {
+    const { orderFormId, salesChannel } = this.props
+    const {
+      bestPickupOptions,
+      pickupPoints,
+      pickupOptions,
+      logisticsInfo,
+    } = this.state
+
+    const pickupAddress = pickupPoint.pickupStoreInfo
+      ? pickupPoint.pickupStoreInfo.address
+      : pickupPoint.address
+
+    if (pickupPoint.pickupStoreInfo) {
+      this.setState({
+        selectedPickupPoint: pickupPoint,
+        activeSidebarState: DETAILS,
+      })
+      return
+    }
+
+    this.setActiveSidebarState(SEARCHING)
+    this.setState({ isSearching: true })
+
+    getAvailablePickup({
+      logisticsInfo,
+      salesChannel,
+      orderFormId,
+      pickupAddress,
+    }).then(data => {
+      const availablePickupOptions =
+        data.logisticsInfo && getPickupOptions(data.logisticsInfo)
+      const availablePickupPoints = data.pickupPoints
+      const availablePickupLI =
+        data.logisticsInfo &&
+        data.logisticsInfo.find(li => this.findSla(li, pickupPoint))
+      const availablePickupSLA =
+        availablePickupLI && this.findSla(availablePickupLI, pickupPoint)
+
+      const availablePickupOptionsWithoutDistance = availablePickupOptions.map(
+        option => ({
+          ...option,
+          pickupDistance: null,
+        })
+      )
+      const availablePickupPointsWithoutDistance = availablePickupPoints.map(
+        option => ({
+          ...option,
+          distance: null,
+        })
+      )
+
+      const newPickupOptions = uniqBy(
```

Comment:
These next 3 constants could use a single logic in a function, that would be easier to test and reuse it later on.